### DISCUSSION TITLE
Toggle/Expand Folder container

### DISCUSSION THREAD

**Discussion by suhaimim** [directus/directus#9939] [Archive]
How can we toggle collapse/expand the folder that has lots of containers/items?

![image](https://user-images.githubusercontent.com/35832684/142721523-a3856426-ba5a-4015-92a3-c2d7b596b1a6.png)

**Comment by rijkvanzanten** [comment]
You can click the folder icon in the screenshot to toggle between "start open" "start closed" or "keep it open" 👍🏻

**Reply by benhaynes** [reply]
This is the correct answer, although I agree that this action is not very clear. I think we should move this to the collection context menu, so when you'd have:

```
Force Open
[Start Open] (Default)
Start Closed
----
Make Collection Hidden
Delete Collection
```

**Reply by suhaimim** [reply]
When I click the left folder icon for Postal Address, this will appear

![image](https://user-images.githubusercontent.com/35832684/142742976-3e55b6d1-15c0-475e-8939-c8ac127ec4a3.png)

![image](https://user-images.githubusercontent.com/35832684/142742970-25925c0d-8c6d-4d89-acc1-a9dd0c6ca694.png)

When I click the right folder icon of Postal Address, only this changed

![image](https://user-images.githubusercontent.com/35832684/142742995-6cb1018f-5d3b-4319-9a51-8e4f8dfb051a.png)

![image](https://user-images.githubusercontent.com/35832684/142743007-2b0ebdc3-cbd1-4adf-86b0-02296c826912.png)

But, both did not collapse (close hide) the collections list.

**Reply by benhaynes** [reply]
It doesn't collapse it in settings, it only collapses it in the actual user menu within the content module. There's currently no way to collapse it within settings.

**Reply by suhaimim** [reply]
OK, understand. Thank you.
Glad if there is a feature to toggle this cause the list can be a lots to scroll in the future.

**Comment by adiazc9** [comment]
@benhaynes Do you have any request for this feature?, so ve can vote for it? It'd be great if we can also collapse the folder on the Data Mosel side, It' be useful for large projects!

**Reply by benhaynes** [reply]
Beyond this feature request, any new one can be opened as well... No others that I'm aware of.

**Comment by that1matt** [comment]
We were running into this issue as well, I created an example of adding the ability to toggle collection items, that also use the folder settings to set them initially.

[Working Example](https://github.com/that1matt/directus/tree/feature-add-collapsable-data-model-items).

It was relatively simple to implement, but I am also new at Vue, the only issue that is present, is the transition-expand doesn't seem to work on the collection items that are collapsing/expanding.

Here is a video of the feature change.

https://user-images.githubusercontent.com/9679004/236709235-8a47a239-2060-45b1-a09c-d1a1b593a307.mov

Hopefully this can be added to directus, as we have tons of collections that make the list hard to navigate.

**Reply by rijkvanzanten** [reply]
Very useful addition! Quick question: should we add some sort of way to communicate how many collections a collapsed parent includes? I'm worried that it might be easy to miss that a collection has hidden/collapsed children 🤔 (cc @benhaynes for UX)

**Reply by rijkvanzanten** [reply]
Opened https://github.com/directus/directus/issues/18517 to track the implementation of this improvement 👍🏻

**Reply by benhaynes** [reply]
I agree that these should have the toggle to expand... the three-way toggle for the "starting state" should probably move into the "..." menu with horizontal rules grouping it and a check next to the one that is active.

**Reply by that1matt** [reply]
I agree that having an indicator that there are nested collections would be ideal.

The code is really in a mvp state, while I learn Vue and Directus. I also want to add some tests around new features like this added.
